src/NzbDrone.Core/Download/ProcessDownloadDecisions.cs

```diff
@@ -36,52 +39,65 @@ public ProcessedDecisions ProcessDecisions(List<DownloadDecision> decisions)
             var prioritizedDecisions = _prioritizeDownloadDecision.PrioritizeDecisions(qualifiedReports);
             var grabbed = new List<DownloadDecision>();
             var pending = new List<DownloadDecision>();
+            var failed = new List<DownloadDecision>();
+
+            var usenetFailed = false;
+            var torrentFailed = false;
 
             foreach (var report in prioritizedDecisions)
             {
                 var remoteEpisode = report.RemoteEpisode;
+                var downloadProtocol = report.RemoteEpisode.Release.DownloadProtocol;
 
-                var episodeIds = remoteEpisode.Episodes.Select(e => e.Id).ToList();
-
-                //Skip if already grabbed
-                if (grabbed.SelectMany(r => r.RemoteEpisode.Episodes)
-                                .Select(e => e.Id)
-                                .ToList()
-                                .Intersect(episodeIds)
-                                .Any())
+                // Skip if already grabbed
+                if (IsEpisodeProcessed(grabbed, report))
                 {
                     continue;
                 }
 
                 if (report.TemporarilyRejected)
                 {
-                    _pendingReleaseService.Add(report);
+                    _pendingReleaseService.Add(report, PendingReleaseReason.Delay);
                     pending.Add(report);
                     continue;
                 }
 
-                if (pending.SelectMany(r => r.RemoteEpisode.Episodes)
-                        .Select(e => e.Id)
-                        .ToList()
-                        .Intersect(episodeIds)
-                        .Any())
+                if (downloadProtocol == DownloadProtocol.Usenet && usenetFailed ||
+                    downloadProtocol == DownloadProtocol.Torrent && torrentFailed)
                 {
-                    continue;
+                    failed.Add(report);
                 }
 
                 try
                 {
                     _downloadService.DownloadReport(remoteEpisode);
                     grabbed.Add(report);
                 }
-                catch (Exception e)
+                catch (DownloadClientException ex)
+                {
+                    if (ex is DownloadClientUnavailableException || ex is DownloadClientAuthenticationException)
```

Comment:
Since an `DownloadClientAuthenticationException` means we failed to connect properly we should also store those releases I think. Both `DownloadClientUnavailableException` and `DownloadClientAuthenticationException`  inherit `DownloadClientException`.